lib/api/src/entities/engine/mod.rs

```diff
@@ -0,0 +1,126 @@
+//! Defines the [`Engine`] type, the [`EngineLike`] trait for implementors and useful
+//! traits and data types to interact with them.
+
+use bytes::Bytes;
+use std::{path::Path, sync::Arc};
+use wasmer_types::DeserializeError;
+
+#[cfg(feature = "sys")]
+use wasmer_compiler::Artifact;
+
+use crate::{IntoBytes, Store};
+
+/// Create temporary handles to engines.
+mod engine_ref;
+
+/// The actual (private) definition of the engines.
+mod inner;
+pub(crate) use inner::RuntimeEngine;
+
+pub use engine_ref::*;
+
+static ENGINE_ID_COUNTER: std::sync::atomic::AtomicU64 = std::sync::atomic::AtomicU64::new(0);
```

Comment:
nit, but I'd probably move the static and the next() function to be statics in the Engine struct impl.

Reply:
[8e5ce03](https://github.com/wasmerio/wasmer/pull/5243/commits/8e5ce03a184cd0851ec4709e9cfdb6312d6726db)